### PR TITLE
Make logging false by default

### LINKED ISSUE #1973: logging.basicConfig should not be called by 'import bokeh'
The `logging.basicConfig` call is typically performed by the user. It is allowed to be called only once during program life time, all subsequent calls do nothing.

For these reason I think it is not nice to have it done implicitly by `import bokeh`. It breaks any application using `bokeh` that configures logging after loading used modules.

The respective line in the code: https://github.com/bokeh/bokeh/blob/master/bokeh/__init__.py#L12)

### PR DESCRIPTION
**Pull request by Maggie-M**
issues: closes #1973

### CODE REVIEW

File: sphinx/source/docs/dev_guide/notes.rst [hunk 1]
```diff
@@ -44,6 +44,9 @@ There are several environment variables that can be useful for developers:
 * ``BOKEH_PRETTY`` --- Whether to emit "pretty printed" JSON
     Accepted values are ``yes``/``no``, ``true``/``false`` or ``0``/``1``.
 
+* ``BOKEH_PY_LOGGING`` -- Whether to enable logging by default.
```
Comment (bryevdv) [inline sphinx/source/docs/dev_guide/notes.rst]: needs three `-` to make an "em" dash like the others also the others don't have a period (probably should but for now just want to be consistent)

File: bokeh/__init__.py [hunk 2]
```diff
@@ -8,10 +9,11 @@
...
 
 # configure logger level
-level = settings.py_log_level()
-logging.basicConfig(level=level)
-# set up the logger
-log = logging.getLogger(__name__)
+if settings.py_logging():
+    level = settings.py_log_level()
+    logging.basicConfig(level=level)
+    # set up the logger
```
Comment (bryevdv) [inline bokeh/__init__.py]: this comment unnecessary

**Comment by bryevdv** [comment]
Few small comments, otherwise LGTM

File: bokeh/__init__.py [hunk 3]
```diff
@@ -1,6 +1,7 @@
 from __future__ import absolute_import, print_function
 
 import logging
+import sys
```
Comment (damianavila) [inline bokeh/__init__.py]: why `sys` here?

Comment (Maggie-M) [inline bokeh/__init__.py]: Ah, thanks for pointing that out!  I had left that in for testing purposes initially and had forgotten it.  I'll remember to run pyflakes on files changed in the future.

**Comment by damianavila** [comment]
LGTM now...

**Pull request merged**